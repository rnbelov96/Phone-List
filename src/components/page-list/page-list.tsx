import React, { useMemo } from 'react';
import { connect } from 'react-redux';
import { AppActionCreators } from '../../reducer/app/app';

export type PageListProps = {
  pageList: number[];
  currentPage: number;

  setCurrentPage: (page: number) => void;
};

export const PurePageList: React.FunctionComponent<PageListProps> = ({
  pageList,
  currentPage,
  setCurrentPage,
}: PageListProps) => {
  const bodyEl = useMemo(() => document.body, []);
  return (
    <div className="page-list">
      {pageList.map((page, index) => (
        <button
          key={`${page}-${index + 1}`}
          className={`page-list__button ${
            page === currentPage ? 'page-list__button_current' : ''
          } ${page === 0 ? 'page-list__button_inactive' : ''}`}
          type="button"
          onClick={() => {
            if (page === currentPage || page === 0) {
              return;
            }
            setCurrentPage(page);
            bodyEl.scrollIntoView({ behavior: 'smooth' });
          }}
        >
          {page === 0 ? '...' : page}
        </button>
      ))}
    </div>
  );
};

export default connect(() => ({}), {
  setCurrentPage: AppActionCreators.setCurrentPage,
})(PurePageList);
