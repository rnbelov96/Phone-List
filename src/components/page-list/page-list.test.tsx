import React from 'react';
import renderer from 'react-test-renderer';
import { PurePageList } from './page-list';

test('Page List should render correctly', () => {
  const tree = renderer
    .create(
      <PurePageList
        currentPage={1}
        pageList={[1, 2, 3, 4, 5, 6, 0, 81]}
        setCurrentPage={() => {}}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
