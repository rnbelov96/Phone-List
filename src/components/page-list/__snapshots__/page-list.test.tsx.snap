// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page List should render correctly 1`] = `
<div
  className="page-list"
>
  <button
    className="page-list__button page-list__button_current "
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="page-list__button  "
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="page-list__button  "
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="page-list__button  "
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="page-list__button  "
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="page-list__button  "
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="page-list__button  page-list__button_inactive"
    onClick={[Function]}
    type="button"
  >
    ...
  </button>
  <button
    className="page-list__button  "
    onClick={[Function]}
    type="button"
  >
    81
  </button>
</div>
`;
